fetcher/utils.go

```diff
@@ -39,3 +39,17 @@ func parseToGauge(gauge prometheus.Gauge, value string) {
 	}
 	gauge.Set(parsed)
 }
+
+func parseAdditionalLabels(additionalLabels []string, labels map[string]string) (result []string) {
+alLoop:
+	for _, al := range additionalLabels {
+		for k, v := range labels {
+			if k == al {
+				result = append(result, v)
+				continue alLoop
```

Comment:
This took me a while to figure out, mainly because I am not used to reading code with jump labels, I guess 🤔.
I know that this practice is not as frowned upon as it is in other languages and it does make the code a bit simpler.

I think for me personaly, I would prefer extracting another function for the inner loop or introduce another variables. If you find this ugly, I am also willing to getting used to having jump labels in the code base, though 😉.

Reply:
Changed with https://github.com/jangraefen/hcloud-pricing-exporter/pull/48/commits/c62f5a807ae70581c6c1d69cdbfce5c8aff732de